### PR TITLE
HPCC-17923 DOCS:ECLWatch Common Elements

### PR DESCRIPTION
**Pull request by g-pan**
Fix HPCC-17923 DOCS:ECLWatch Common Elements
Add chapter (or section(s)) to ECL Watch to
detail common elements in ECL Watch which work
universally throughout the interface.
Also addresses HPCC-14064

@JamesDeFabia please review
@miguelvazq please review

Signed-off-by: G-Pan <greg.panagiotatos@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [X] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 1]
```diff
@@ -63,20 +66,448 @@
...
+    items with common functionality universally across pages of ECL
+    Watch.</para>
+
+    <sect1 id="Common_Elements">
+      <title>Common Elements</title>
+
+      <para>There are several common elements throughout the pages of ECL
+      Watch. These are very useful features that are available throughout the
+      various pages of ECL Watch. These common elements can work anywhere
+      within ECL Watch.</para>
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: Replace 3 sentences with this one:
There are common elements that are useful anywhere within ECL Watch.

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 2]
```diff
@@ -63,20 +66,448 @@
...
+            <mediaobject>
+              <imageobject>
+                <imagedata fileref="images/ECLWA500.jpg" />
+              </imageobject>
+            </mediaobject>
+          </figure></para>
+
+        <para>Opening a new window directly into the view you have navigated
+        to can be useful in many ways: </para>
+
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: Opening a new page of the current view is useful in several ways:

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 3]
```diff
@@ -63,20 +66,448 @@
...
+        <itemizedlist>
+          <listitem>
+            <para>Open in new page is a shareable link. You can right-click on
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: Open in New Page

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 4]
```diff
@@ -63,20 +66,448 @@
...
+            Open in New Page and select <emphasis>Copy link address</emphasis>
+            from the drop list. You can now share or send that link and the
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: from the drop list
s/b
from the context menu

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 5]
```diff
@@ -63,20 +66,448 @@
...
+            recipient can open a window directly to the (new) page that you
+            shared. </para>
+          </listitem>
+
+          <listitem>
+            <para>The new pages maintain filter states, so that if you had set
+            a filter for jobs from a specific user, then shared that new page
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: New pages maintain filter states. If you have set a filter and share a link, the new page has the same filter set.

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 6]
```diff
@@ -63,20 +66,448 @@
...
+          </figure></para>
+
+        <para>ECL Watch commonly displays items in the forms of a list. Lists
+        of workunits, lists of logical files, lists of Roxie queries, etc. The
+        following elements in ECL Watch work on any list widget:</para>
+
+        <para><itemizedlist>
+            <listitem>
+              <para>Download as CSV (Workunits, DFU Workunits, Logical
+              files)</para>
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: , and Roxie queries

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 7]
```diff
@@ -63,20 +66,448 @@
...
+        <sect3>
+          <title>List Filters</title>
+
+          <para>Filter options on a list allow you to use specified criteria
+          to set what you display in the list. Press the <emphasis
+          role="bold">Filter</emphasis> action button to display the filter
+          options. When you specify any Filter options, the Filter action
+          button changes and then displays <emphasis role="bold">Filter
+          Set</emphasis> and the filter indicator image lights up. When you
+          then copy an <emphasis>Open new Page</emphasis> link with the filter
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: s/b role="bold" and say: Open in New Page

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 8]
```diff
@@ -63,20 +66,448 @@
...
+          set, the filter condition persists. Filter conditions continue to
+          refine the results displayed until you clear the filter.</para>
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: Filter conditions limit the results displayed until you clear the filter.

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 9]
```diff
@@ -63,20 +66,448 @@
...
+
+      <sect2 id="ECLWA_BannerItems">
+        <title>Banner Items</title>
+
+        <para>These are the items along the top banner area of the main ECL
+        Watch page. Several of these items are detailed either later in this
+        document or in a separate document, such a the systems monitor
+        indicator light. That is touched on briefly here but for more
+        information refer to the HPCC Monitoring and Reporting doc.</para>
+
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: This paragraph needs to be worded better

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 10]
```diff
@@ -63,20 +66,448 @@
...
+          in subsequent chapters.</para>
+        </sect3>
+
+        <sect3 id="MonitorIndicatorLight">
+          <title>Monitor indicator light</title>
+
+          <para>The Monitor indicator light is used to provide an at a glance
+          snapshot of your overall system health. If you have monitoring and
+          reporting configured (see HPCC Monitoring and Reporting) then this
+          light will display a color based on the the overall (configurable)
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: HPCC Monitoring and Reporting s/b italics

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 11]
```diff
@@ -63,20 +66,448 @@
...
+          system health. The light is black if there is no monitoring or
+          reporting configured for your system.</para>
+        </sect3>
+
+        <sect3 id="DownloadAsCSV">
+          <title>Download as CSV</title>
+
+          <para>With every list in ECL Watch you have the ability to download
+          it as a CSV formatted file. You can then open the CSV file in any
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: ...ability to download selected rows as a ...

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 12]
```diff
@@ -63,20 +66,448 @@
...
+          spreadsheet application which supports CSV input.</para>
+        </sect3>
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: which supports CSV format.

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 13]
```diff
@@ -63,20 +66,448 @@
...
+            <mediaobject>
+              <imageobject>
+                <imagedata fileref="images/ECLWA008.jpg" />
+              </imageobject>
+            </mediaobject>
+          </figure></para>
+
+        <para>You can search ECL Workunits, DFU Workunits, Logical Files, and
+        Queries using the global search box. The global search box also
+        supports wild cards. To limit or filter your search results you can
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: wildcard

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 14]
```diff
@@ -63,20 +66,448 @@
...
+          you to set a custom banner message at the top of your browser window
+          when you open ECL Watch. You can use this feature to send messages
+          to users.</para>
+
+          <para>The <emphasis role="bold">Error/Warning(s)</emphasis> link
+          displays a tab showing you Errors, Warnings, and Information
+          messages. You can filter this page by checking the boxes at the
+          bottom of the tab. A copy facility is also provided.</para>
+
+          <para>The <emphasis role="bold">Open Legacy ECL Watch</emphasis>
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: Open Legacy ECL Watch has been removed.

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 15]
```diff
@@ -63,20 +66,448 @@
...
+          link opens a tab to the previous version of ECL Watch. The legacy
+          version of ECL Watch is deprecated; this link is only available for
+          a limited time to help in the transition.</para>
+
+          <para>The <emphasis role="bold">Additional Resources</emphasis> link
+          opens a tab to the HPCC Systems<superscript>®</superscript> download
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: This has changed. It now has a submenu ...

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 16]
```diff
@@ -63,20 +66,448 @@
...
+                <para>Verify the password expiration date, or if password is
+                set to expire.</para>
+              </listitem>
+            </orderedlist></para>
+
+          <sect4 id="ECLWatch_AdvancedMenuChangePassword" role="nobrk">
+            <title>Change Password</title>
+
+            <para>If authentication is enabled on your HPCC system, you can
+            change your password, right from the User Details window. <!--***? is this still valid : don't see where this is in new ECLWatch ***--><orderedlist>
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: This is still valid, you can remove comment

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
Comments inline

**Comment by g-pan** [comment]
Updates made. @JamesDeFabia please review.

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 17]
```diff
@@ -113,27 +111,27 @@
...
-            a filter for jobs from a specific user, then shared that new page
-            link, the filters are still set.</para>
+            <para>New pages maintain filter states. If you have set a filter
+            and share that link, the new page has the same filter set.</para>
           </listitem>
 
           <listitem>
             <para>The new page is live. The recipient doesn't need to reset
-            the page or refresh to see current state. </para>
+            the page or refresh to see current state.</para>
```
Comment (miguelvazq) [inline docs/ECLWatch/TheECLWatchMan.xml]: It's live **if** there is an auto-refresh mechanism in place.

File: docs/ECLWatch/TheECLWatchMan.xml [hunk 18]
```diff
@@ -190,22 +188,18 @@
...
         <title>Banner Items</title>
 
-        <para>These are the items along the top banner area of the main ECL
-        Watch page. Several of these items are detailed either later in this
-        document or in a separate document, such a the systems monitor
-        indicator light. That is touched on briefly here but for more
-        information refer to the HPCC Monitoring and Reporting doc.</para>
-
-        <para>The Banner Items are a constant in the interface. The top banner
+        <para>The banner area of ECL Watch appears at the top of all pages.
```
Comment (JamesDeFabia) [inline docs/ECLWatch/TheECLWatchMan.xml]: The banner area of ECL Watch appears at the top of all pages. The Banner Items are a constant in the interface. The top banner remains a fixture at the top of your window as you navigate through the pages.

3 sentences when one will do. Keep "The banner area of ECL Watch appears at the top of all pages."

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]
comments inline

**Review by JamesDeFabia — APPROVED** [review]
Looks good now

**Comment by g-pan** [comment]
Revised, updated, and squashed for a clean merge.

**Pull request merged**